### PR TITLE
Adding tests to 'contacts' app

### LINKED ISSUE #64: Add test via PyTest to 'Contacts' app
Tests via pytest are required for proper testing of our code.
Expected outcome:
- Making a connection - SUCCESS
- Making a connection between Seeker & Seeker - FAIL
- Making a connection that already exists - FAIL

### PR DESCRIPTION
**Pull request by DanielMy777**
- Adding tests via pytest[1] to the contacts app, focusing on tests for the
connection model.
- Changing the Apartment model import in the test data file.
List of tests is stated in the commit log.
fix #64 
[1]: https://docs.pytest.org/en/6.2.x/

### CODE REVIEW

File: contacts/tests.py
Comment (Micha546) [inline contacts/tests.py]: Why did you add all those create functions? what is their purpose?

**Comment by OrAsaf** [comment]
@DanielMy777 since this is a testing PR I would suggest you rebase it so the purest can run and we can see in the PR that the tests work as intended.

File: contacts/tests.py
Comment (Micha546) [inline contacts/tests.py]: create_user is already a function in User, maybe change the name to avoid a potential conflict

File: contacts/tests.py
Comment (DanielMy777) [reply contacts/tests.py]: There's a strong purpose,
I have to create multiple objects in order to create connections.
Those include creating an apartment, seeker, and each requires creating a user too.

File: contacts/tests.py
Comment (DanielMy777) [reply contacts/tests.py]: Noted, I'll change the name

**Comment by DanielMy777** [comment]
@1Oreo I've rebased, fixed some stuff that were hurt during the last merges, and the tests run great now.
Waiting for your further review :)

**Comment by tamirho** [comment]
LGTM

File: contacts/tests.py
Comment (OrAsaf) [inline contacts/tests.py]: This is a bit ambiguous. You send connection_1 to the function but don't use it, I think I know why you are doing just that but I would like to hear the reason please.

Comment (DanielMy777) [reply contacts/tests.py]: @1Oreo Connection_1 is created as a pytest fixture, it saves a row in the db with a connection between seeker_1 and apartment_1.
In the test, I want to save *another* connection row between seeker_1 and apartment_1, which should cause an IntegrityError. To make that first original connection be inserted to the db I had to send connection_1 to the function. It's a fixture so as long as I don't add it as an argument, it's like that row was never added in the first place (will not activate the fixture and therefore will not save connection_1). Causing the "second" row that was added in the test function to be the first and only one, which will not cause a duplication error.

Comment (OrAsaf) [reply contacts/tests.py]: So I figured this is what you meant to create.
But the way you implemented it might cause someone to misunderstand the intention. What I would do to sort it out is use `connection_1.seeker` and `connection_1.appartment` so that when someone reads the code he will understand that you are re-inserting the same connection to cause an error.

Comment (DanielMy777) [reply contacts/tests.py]: @1Oreo You got me 🎯 
I took your tip and changed it, let me know what you think :)

**Review by OrAsaf — APPROVED** [review]

File: contacts/tests.py
Comment (Omeramsc) [inline contacts/tests.py]: You can achieve the same using one fixture Instead of creating two, almost identical ones
for example: 
```
@pytest.fixture
def fixture_with_different_results():
    def _inner_function(par_1, par_2, par_3):
        changing_list = [par_1,par_2,par_3]
        return changing_list

    return _inner_function

def test_the_fixture(fixture_with_different_results):
    first_item = fixture_with_different_results(1,2,3)
    second_item = fixture_with_different_results(4,5,6)
```

first_item will get a list containing [1,2,3]
second_item will get a list containing [4,5,6]

Comment (Omeramsc) [reply contacts/tests.py]: It'll also give you more flexibility with the information you want to insert for the test.

File: contacts/tests.py
Comment (Omeramsc) [inline contacts/tests.py]: If there's no `apartment_2` or something of that sort (and applying my former comment should remove the need for those), please name the fixtures in a better way (this comment applies to all of them).

File: contacts/tests.py
Comment (Omeramsc) [inline contacts/tests.py]: You can apply the fixture logic I mentioned for those functions as well.

**Review by Omeramsc — CHANGES_REQUESTED** [review]

File: contacts/tests.py
Comment (DanielMy777) [reply contacts/tests.py]: @Omeramsc Wouldn't that require me to call this fixture and create a seeker inside the test function?
I thought that the main point of a fixture is to have blocks of code happen before the test starts, that's what I got from Edward. For example, if I understand correctly you'd like something like:
```
def test_invalid_users_connection(fixture_with_different_results):
    seeker_1 = fixture_with_different_results(*seeker1 data*)
    seeker_2 = fixture_with_different_results(*seeker2 data*)
    with pytest.raises(ValueError):
        with transaction.atomic():
            Connection(seeker=seeker_1, apartment=seeker_2).save()
```
There's other rows I neglected as well (creating the user, which is part of the seeker data, before calling the fixture), Which will add lines of code and complexity IMO.

File: contacts/tests.py
Comment (DanielMy777) [reply contacts/tests.py]: Thanks, I renamed the fixtures and marking this as resolved!

File: contacts/tests.py
Comment (Omeramsc) [reply contacts/tests.py]: Shorter isn't always better or less complex; While this feature of fixtures is great, I personally wouldn't create two separate fixture for entries as similar as those. @simzacks @1Oreo  WDYT?

Comment (OrAsaf) [reply contacts/tests.py]: I agree with what @Omeramsc said. With the way the project is now (small and in the early days) writing a fixture for every fixture can work, and still be readable and understandable, but in a much bigger application it will again create ambiguity. One of the traits good code needs is to be readable but also understandable and I think what Omer suggested shows someone who reads the code that the fixture is of the same type, just with 2 different users, and also "saves" duplicating code.

Comment (DanielMy777) [reply contacts/tests.py]: @Omeramsc I've never seen this and couldn't really find it online, but gave it a shot.
I changed to code, please tell me if this was what you meant, Thanks!

File: contacts/tests.py
Comment (Omeramsc) [inline contacts/tests.py]: Please refrain from mentioning "fixture" in the name, as it is already declared in the decorator. 
a function's name is important and has a huge role in readability, so always pay attention to it.

File: contacts/tests.py
Comment (Omeramsc) [inline contacts/tests.py]: _make_seeker*

File: contacts/tests.py
Comment (Omeramsc) [inline contacts/tests.py]: _make_seeker*

**Review by Omeramsc — CHANGES_REQUESTED** [review]

File: contacts/tests.py
Comment (Omeramsc) [reply contacts/tests.py]: This is called factories as fixtures, you can read more about it (and pytest fixtures) here:
https://docs.pytest.org/en/stable/fixture.html#factories-as-fixtures

I left some comments, but it already looks way better. good job. :)

File: contacts/tests.py
Comment (DanielMy777) [reply contacts/tests.py]: I removed the 'fixture' part and called the function 'make_seeker'

File: contacts/tests.py
Comment (DanielMy777) [reply contacts/tests.py]: I renamed the inner function '_make_seeker' as you suggested

File: contacts/tests.py
Comment (DanielMy777) [reply contacts/tests.py]: As these aren't fixtures, I'm having trouble understanding what you mean. I've tried some things but to no avail, can you please give me an example?
(The fixture logic applied above uses these functions to create objects in the models)

File: contacts/tests.py
Comment (DanielMy777) [reply contacts/tests.py]: @Omeramsc Thanks! Glad to know that this looks better.
I fixed the code due to your comments, please review again and let me know if there's anything else I should change.

Comment (DanielMy777) [reply contacts/tests.py]: Also, pytest won't work until #91  & #92  are merged so please ignore it while reviewing.

File: contacts/tests.py
Comment (Omeramsc) [inline contacts/tests.py]: This can be included in the fixture - no need for a separate function.

File: contacts/tests.py
Comment (Omeramsc) [inline contacts/tests.py]: We've got a fixture doing the exact same thing, and this is never getting used, you can remove it.

File: contacts/tests.py
Comment (Omeramsc) [inline contacts/tests.py]: same as above

File: contacts/tests.py
Comment (Omeramsc) [reply contacts/tests.py]: I meant that if you're already going the fixture road, you can make `create_new_user` into a fixture as well. fixtures can use other fixtures, so you should not have issues applying it in the current situation.

**Review by Omeramsc — CHANGES_REQUESTED** [review]

File: contacts/tests.py
Comment (DanielMy777) [reply contacts/tests.py]: Included in a fixture

File: contacts/tests.py
Comment (DanielMy777) [reply contacts/tests.py]: You're right it was old, didn't notice. Removed.

File: contacts/tests.py
Comment (DanielMy777) [reply contacts/tests.py]: Included in a fixture.

**Comment by DanielMy777** [comment]
@Omeramsc I think I finally got what you meant now, please review my changes and let me know what you think :)

**Review by Omeramsc — APPROVED** [review]
Looking much better! I'm glad to see you got it.

**Pull request merged**
